pkg/controller/jobframework/reconciler.go

```diff
@@ -621,7 +624,13 @@ func (r *JobReconciler) getAncestorJobManagedByKueue(ctx context.Context, jobObj
 		if QueueNameForObject(parentJob) != "" {
 			return parentJob, nil
 		}
-		jobObj = parentJob
+		currentJob = parentJob
+		linksTraversed++
+		if linksTraversed > managedOwnersChainLimit {
```

Comment:
```suggestion
		if seen.Len() > managedOwnersChainLimit {
```

Do we need to maintain `linksTraversed`? Instead of that, could we just count `seen`?

Reply:
good point.  Done.